docker/docker-compose.yml

```diff
@@ -17,9 +17,9 @@ services:
           "CMD",
           "node",
           "-e",
-          "require('http').get('http://localhost:3000/api/profile', (r) => {if (r.statusCode !== 200) throw new Error(r.statusCode)})"
+          "require('http').get('http://studio:3000/api/profile', (r) => {if (r.statusCode !== 200) throw new Error(r.statusCode)})"
         ]
-      timeout: 5s
+      timeout: 10s
```

Comment:
**remark**

Noticed that this heahtcheck for some reason is very slow (maybe node starting time) on my tests it was taking ~6-7s to return OK.

Reply:
Yes when I tested the healthcheck, it took a few seconds to spawn the nodejs process, really slow.
I also had to change the host for the healthcheck to work.
Currently I'm running with the changes from https://github.com/supabase/supabase/pull/27487

Reply:
Yes, I finally understood where the issue was with those hosts.

On the `studio` the issue lies into the built `server.js` file that contain this:

```
root@be346e9b78f9:/app# cat apps/studio/server.js 
const path = require('path')

const dir = path.join(__dirname)

process.env.NODE_ENV = 'production'
process.chdir(__dirname)

const currentPort = parseInt(process.env.PORT, 10) || 3000
const hostname = process.env.HOSTNAME || '0.0.0.0'

let keepAliveTimeout = parseInt(process.env.KEEP_ALIVE_TIMEOUT, 10)
```

Here we can see that the server is actually ran under `HOSTNAME` if defined in the env, which is the case within the `docker-compose` network container. 

Running `netstat` confirm that the server binding is over the docker alias and not `0.0.0.0` which explain why, within the container, trying to `curl http://studio:3000/api/profile` will work while `curl http://127.0.0.1:3000` or `curl http://localhost:3000` will fail.

And I think the issue is similar for the `storage` container even tough the "server" start file is much less readable, we can see that the actual host that we bind onto `0.0.0.0` and while this is similar for what we have for the `analytics` container, we can see in this one that the issue is that while `analytics` use `curl` which automatically alias `localhost` to `127.0.0.1` which match `0.0.0.0`, `wget` doesn't do the same. And will fail if you don't provide `0.0.0.0` or `127.0.0.1`

```
/opt/app/rel/logflare/bin # netstat -tuln | grep 4000
tcp        0      0 0.0.0.0:4000            0.0.0.0:*               LISTEN      
/opt/app/rel/logflare/bin #  wget --no-verbose --tries=1 --spider "http://localhost:4000/health"
Connecting to localhost:4000 ([::1]:4000)
wget: can't connect to remote host: Connection refused
/opt/app/rel/logflare/bin #  wget --no-verbose --tries=1 --spider "http://0.0.0.0:4000/health"
Connecting to 0.0.0.0:4000 (0.0.0.0:4000)
remote file exists
/opt/app/rel/logflare/bin #  curl -v "http://localhost:4000/health"
* processing: http://localhost:4000/health
*   Trying [::1]:4000...
* connect to ::1 port 4000 failed: Connection refused
*   Trying 127.0.0.1:4000...
* Connected to localhost (127.0.0.1) port 4000
> GET /health HTTP/1.1
> Host: localhost:4000
> User-Agent: curl/8.2.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< cache-control: max-age=0, private, must-revalidate
< content-length: 112
< content-type: application/json; charset=utf-8
< date: Tue, 26 Nov 2024 15:25:38 GMT
< logflare-node: logflare@127.0.0.1
< server: Cowboy
< x-request-id: GAuOHZA13KjXRaIAACui
< 
* Connection #0 to host localhost left intact
{"nodes": ["logflare@127.0.0.1"],"nodes_count":1,"proc_count":804,"status":"ok","this_node":"logflare@127.0.0.1"}
```

We can see a similar setup with the `vector` container, where we can see that the `wget` health-check require to use the container name alias rather than the `localhost`.

@sweatybridge I see two ways to go with this, either we fix the underlying server builds to always bind to `0.0.0.0` but we will still need to change the `wget` healthchecks to be performed on either `127.0.0.1` or `0.0.0.0` rather than localhost. Or we can do like we currently do for the `vector` container, and use the container network alias as the static value that we can be sure should work the same across commands.

Reply:
Using `auth` hostname sounds good to me. Thanks for digging into the root cause for this.